### PR TITLE
Huggingface configurations refactoring

### PR DESCRIPTION
**Pull request by sindhuvahinis**
Tests:
Added unit test cases.
Tested CI test cases manually in ec2. And also ran in github actions
HF https://github.com/deepjavalibrary/djl-serving/actions/runs/6790943108/job/18461535915
HF lora https://github.com/deepjavalibrary/djl-serving/actions/runs/6791093901/job/18462088800
Rolling batch changes https://github.com/deepjavalibrary/djl-serving/actions/runs/6790720395/job/18460807680

### CODE REVIEW

File: engines/python/setup/djl_python/properties_manager/hf_properties.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    def construct_kwargs_quantize(cls, properties):
+        kwargs = properties['kwargs']
+
+        if 'quantize' in properties and not properties['quantize']:
+            return properties
+
+        # device map is not required for lmi dist and
+        if properties['rolling_batch'] == RollingBatchEnum.lmidist or \
+                properties['rolling_batch'] == RollingBatchEnum.vllm:
+            return properties
```
Comment (sindhuvahinis) [inline engines/python/setup/djl_python/properties_manager/hf_properties.py:161]: @xyang16 @KexinFeng @lanking520 
I am not validating whether device_map exists for lmi dist and vllm. And also not inserting load_in_8bit in kwargs. Kindly check if this is okay,

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 import torch
 
-QUANTIZATION_SUPPORT_ALGO = ["bitsandbytes", "gptq"]
+QUANTIZATION_SUPPORT_ALGO = ["bitsandbytes8", "bitsandbytes", "gptq"]
```
Comment (sindhuvahinis) [inline engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py:26]: This is not needed. Let me remove it.

File: engines/python/setup/djl_python/properties_manager/hf_properties.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    # added for backward compatibility lmi-dist
+    bitsandbytes = 'bitsandbytes'
+    gptq = 'gptq'
+
+    # huggingface
+    bitsandbytes4 = 'bitsandbytes4'
+    bitsandbytes8 = 'bitsandbytes8'
+
+    # supported by vllm
+    awq = 'awq'
```
Comment (sindhuvahinis) [inline engines/python/setup/djl_python/properties_manager/hf_properties.py:23]: Kindly verify whether does it cover all supported quantization methods. lmi-dist has tests for gptq. vllm does not have test case for awq in our pipeline. Let me add it.

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py (hunk 2 above)
Comment (sindhuvahinis) [reply engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py]: Resolved.

File: engines/python/setup/djl_python/properties_manager/hf_properties.py (hunk 3 above)
Comment (lanking520) [reply engines/python/setup/djl_python/properties_manager/hf_properties.py:23]: there is no awq test for vLLM, feel free to add it

**Review by lanking520 — APPROVED** [review]

File: engines/python/setup/djl_python/properties_manager/hf_properties.py (hunk 1 above)
Comment (lanking520) [reply engines/python/setup/djl_python/properties_manager/hf_properties.py:161]: device map is not important. this is only applies to scheduler and HF Accelerate path
